### PR TITLE
executor: add trace info for unionScanExec

### LINKED ISSUE #32631: Need trace info in `union scan executor`
## Enhancement
``` mysql
create table t1(a int, b int, c int);
insert into t1 values (1, 1, 1);
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
insert into t1 select * from t1;
create sequence s;
update t1 set a=nextval(s);

create table t(a int, b int, c int, unique key idx_a(a, c));
insert into t values (0, 0, 0);
analyze table t;
begin;
insert into t select * from t1;
explain update t set c=4000 where a=0;
trace format='row' update t set c=4000 where a=1;
```

### The plan
```
mysql> explain update t set c=4000 where a=0;
+-----------------------------+---------+-----------+---------------+------------------+
| id                          | estRows | task      | access object | operator info    |
+-----------------------------+---------+-----------+---------------+------------------+
| Update_5                    | N/A     | root      |               | N/A              |
| └─UnionScan_7               | 1.00    | root      |               | eq(test.t.a, 0)  |
|   └─TableReader_10          | 1.00    | root      |               | data:Selection_9 |
|     └─Selection_9           | 1.00    | cop[tikv] |               | eq(test.t.a, 0)  |
|       └─TableFullScan_8     | 1.00    | cop[tikv] | table:t       | keep order:false |
+-----------------------------+---------+-----------+---------------+------------------+
5 rows in set (0.00 sec)
```

### The trace info
```
mysql> trace format='row' update t set c=4000 where a=2;
+------------------------------------------------------+-----------------+------------+
| operation                                            | startTS         | duration   |
+------------------------------------------------------+-----------------+------------+
| trace                                                | 16:36:46.380747 | 3.676375ms |
|   └─session.ExecuteStmt                              | 16:36:46.380759 | 3.655709ms |
|     ├─executor.Compile                               | 16:36:46.380776 | 496.416µs  |
|     └─session.runStmt                                | 16:36:46.381304 | 3.085125ms |
|       ├─TableReaderExecutor.Open                     | 16:36:46.381359 | 75.625µs   |
|       │ └─distsql.Select                             | 16:36:46.381384 | 40.167µs   |
|       │   └─regionRequest.SendReqCtx                 | 16:36:46.381552 | 327.041µs  |
|       └─executor.handleNoDelayExecutor               | 16:36:46.384236 | 125.792µs  |
|         └─*executor.UpdateExec.Next                  | 16:36:46.384241 | 102.291µs  |
|           └─*executor.UnionScanExec.Next             | 16:36:46.384254 | 83.542µs   |
|             └─*executor.TableReaderExecutor.Next     | 16:36:46.384259 | 58.208µs   |
+------------------------------------------------------+-----------------+------------+
11 rows in set (0.00 sec)
```

### Expect trace info
```
mysql> trace format='row' update t set c=4000 where a=1;
+------------------------------------------------------+-----------------+------------+
| operation                                            | startTS         | duration   |
+------------------------------------------------------+-----------------+------------+
| trace                                                | 16:33:25.545406 | 8.480583ms |
|   └─session.ExecuteStmt                              | 16:33:25.545473 | 8.396959ms |
|     ├─executor.Compile                               | 16:33:25.545511 | 941.625µs  |
|     └─session.runStmt                                | 16:33:25.546507 | 7.328958ms |
|       ├─UnionScanExec                                | 16:33:25.546647 | 6.835292ms |
|       │ ├─TableReaderExecutor.Open                   | 16:33:25.546650 | 142.542µs  |
|       │ │ └─distsql.Select                           | 16:33:25.546704 | 65.584µs   |
|       │ │   └─regionRequest.SendReqCtx               | 16:33:25.546949 | 535.666µs  |
|       │ └─UnionScanExec.open                         | 16:33:25.546832 | 6.6075ms   |
|       └─executor.handleNoDelayExecutor               | 16:33:25.553599 | 204µs      |
|         └─*executor.UpdateExec.Next                  | 16:33:25.553610 | 152.375µs  |
|           └─*executor.UnionScanExec.Next             | 16:33:25.553627 | 121.333µs  |
|             └─*executor.TableReaderExecutor.Next     | 16:33:25.553638 | 81.709µs   |
+------------------------------------------------------+-----------------+------------+
13 rows in set (0.01 sec)
```

### Problems
In the original trace info, it spends more than 3 ms in `session.runStmt`, but can't find details.

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #32631

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
add trace info for unionScanExec
```

new trace info:
```mysql
mysql> trace format='row' update t  set c=4000 where a=2;
+------------------------------------------------------+-----------------+------------+
| operation                                            | startTS         | duration   |
+------------------------------------------------------+-----------------+------------+
| trace                                                | 17:24:51.772506 | 7.905792ms |
|   └─session.ExecuteStmt                              | 17:24:51.772516 | 7.870125ms |
|     ├─executor.Compile                               | 17:24:51.772555 | 759.375µs  |
|     └─session.runStmt                                | 17:24:51.773378 | 6.972209ms |
|       ├─UnionScanExec.Open                           | 17:24:51.773475 | 6.510208ms |
|       │ ├─TableReaderExecutor.Open                   | 17:24:51.773479 | 151µs      |
|       │ │ └─distsql.Select                           | 17:24:51.773531 | 72.875µs   |
|       │ │   └─regionRequest.SendReqCtx               | 17:24:51.773782 | 659.709µs  |
|       │ ├─buildMemTableReader                        | 17:24:51.773656 | 9.667µs    |
|       │ └─memTableReader.getMemRows                  | 17:24:51.773688 | 6.237834ms |
|       └─executor.handleNoDelayExecutor               | 17:24:51.780123 | 195.333µs  |
|         └─*executor.UpdateExec.Next                  | 17:24:51.780135 | 157.834µs  |
|           └─*executor.UnionScanExec.Next             | 17:24:51.780149 | 126.5µs    |
|             └─*executor.TableReaderExecutor.Next     | 17:24:51.780159 | 84.166µs   |
+------------------------------------------------------+-----------------+------------+
14 rows in set (0.01 sec)
```

### CODE REVIEW

**Comment by Defined2014** [comment]
/cc @tiancaiamao @xhebox @lcwangchao 
PTAL

File: executor/mem_reader.go [hunk 1]
```diff
@@ -15,6 +15,8 @@
 package executor
 
 import (
+	"context"
```
Comment (tangenta) [inline executor/mem_reader.go:18]:
```suggestion
	"context"

```

**Review by xhebox — APPROVED** [review]

**Review by tiancaiamao — APPROVED** [review]

**Pull request merged**